channeld/channel.c

```diff
@@ -169,7 +169,7 @@ static void start_commit_timer(struct peer *peer);
 
 static void billboard_update(const struct peer *peer)
 {
-	const char *funding_status, *announce_status, *shutdown_status;
+	const char *funding_status = NULL, *announce_status = NULL, *shutdown_status = NULL;
 
 	if (peer->funding_locked[LOCAL] && peer->funding_locked[REMOTE])
 		funding_status = "Funding transaction locked.";
```

Comment:
It is this very initialization which suppresses any -O1 compile warning about them being unset (if they were, in some future path).  So there's no great answer here, but either we do the initialization *and* assert, or we do neither.